dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/ConvolutionalLayer.java

```diff
@@ -77,13 +82,87 @@ public boolean isLearnable() {
   }
 
   /**
+   * Transform the given image to column form to facilitate matrix multiplication.
+   * @param imageIndex the index of the image in the input matrix.
+   * @param input input values for this layer.
+   * @return the converted column.
+   */
+  private Matrix im2col(final int imageIndex, final Matrix input) {
+    final int[] inputShape = getInputShape();
+    final Matrix col =
+        matrixFactory.zeros(NeuralNetworkUtils.getShapeLength(outputShape), kernelHeight * kernelWidth);
+    for (int kh = 0; kh < kernelHeight; ++kh) {
```

Comment:
This part seems to have the opportunity to to be optimized.
For example, since our matrix implementation uses column-major order, it might be more efficient to use the loop for `kh` as the innermost loop, which can increase cache hit rate for `input`.